### ISSUE TITLE
Share Console composition and session lifecycle for new and loaded games

### ISSUE THREAD

**Issue by Extender92** [Extender92/Monopoly#18]
## Problem/Goal

New-game and loaded-game paths construct Console dependencies and session state differently, and recursive menu transitions make ownership and subscription cleanup unreliable. Establish one composition and lifecycle path for every match session.

## Required behavior

- New and loaded matches enter the same composition path.
- One explicit application/session loop owns navigation; menus return explicit results instead of invoking one another recursively.
- Menus receive their `IConsoleWrapper`, session services, and stores through composition and never construct their own wrapper or match session.
- Replacing a match and every exit path dispose prior subscriptions and transient session resources exactly once.
- A failed load keeps the current session alive and unchanged.

## Scope

- Console composition root, navigation results, and application/session lifecycle.
- New/load/replace/exit wiring and lifecycle tests.
- Subscription ownership needed by Console.

## Out of scope

- Persistence implementation or Save Format Version 2.
- Save/load prompts beyond the integration seams for #16 and #17.
- Rendering layout or game-rule changes.

## Acceptance criteria

- New and successfully loaded matches use the same dependency graph and session runner.
- No menu calls another menu/session recursively.
- Menus construct no `ConsoleWrapper`, persistence store, or match session internally.
- Navigation outcomes for new, load, resume, cancel, and exit are explicit and exhaustively handled.
- Match replacement and all exit/error paths dispose old subscriptions once and only once.
- A failed/cancelled load leaves the existing match, subscriptions, and current turn unchanged.
- Lifecycle tests cover repeated new/load cycles and prove handler counts do not grow.
- Architecture and Console documentation describe the composition ownership.
- Full Release build and test suite pass without new warnings.

## Dependencies

- Coordinate subscription ownership with match-scoped notifications #50.
- Establish this lifecycle before #16 and #17.

## Related issues

- Related to #20
- Related to #31
- Related to #33

**Comment by Extender92** [comment]
Deferred during neutral-Core backlog sanitation. This issue has not been implemented. It will not be copied to the clean repository and is not a cutover prerequisite. Reconsider the behavior only as a new, independently scoped neutral issue after cutover, based on the clean architecture and current product goals.
